Sources/GRPCCodeGen/PipelineStage.swift

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+struct PipelineStage<Input, Output> {
+  private(set) var transition: (Input) throws -> Output
+
+  func run(_ input: Input) throws -> Output {
+    return try transition(input)
+  }
+}
```

Comment:
I know this doesn't make sense (yet). I want to check if this is the right direction (if I should have stages like this - and extend the struct with other useful properties or not).

Reply:
I don't think having the pipeline is wrong - I wonder if we've got enough steps right now to warrant it or if it's unnecessary, but it does maybe help to organise the code more neatly, or make it more obvious what it is we're doing each step.

However, I think defining different properties in `Pipeline.swift` for each stage and calling them manually defeats the purpose of having a pipeline architecture. I think each step should be doing their bit and then handing over its result to the next stage - and then return that result. This is similar to how interceptors work in gRPC or handlers in NIO.